18-websockets/classworks/Anna_Volodkova/web/src/pages/Chat/Chat.jsx

```diff
@@ -1,45 +1,51 @@
-import React, {useState} from 'react';
-import './Chat.css';
+import React, { useCallback, useState, useEffect } from 'react';
+import { debounce } from 'lodash';
+
 import Message from './components/Message';
+import './Chat.css';
 
 function Chat(props) {
   const socket = props.socket;
   const [value, setValue] = useState('');
   const [userName, setUserName] = useState('');
-
-  const [isTyping, setIsTyping] = useState(false);
-  const [typingUser, setTypingUser] = useState('')
-
   const [messages, setMessages] = useState([]);
+  const [typingUsers, setTypingUsers] = useState([]);
 
-  socket.on('get-message', data => {
-    setMessages([...messages, data]);
-  });
-
-  if(isTyping){
-    socket.emit('send-isTyping', userName);
+  const setTypingUsersNames = (name) => {
+    if(!typingUsers.includes(name)){
+      setTypingUsers([...typingUsers, name]);
+      console.log('TYPING USERS AND NAME', typingUsers, name);
+      setTimeout(() => {
```

Comment:
you will create a timeout for every typing event. Your typing message can disappear early.
I'd recommend reset the timer if user is already in `setTypingUsers` array.